src/retry.ts

```diff
@@ -34,10 +34,11 @@ export function retry<T extends Function>(fn: T, retryOpts: RetryOpts): T {
       try {
         return await fn(...args);
       } catch (err) {
-        if (retryOpts.isRetryable && !retryOpts.isRetryable(err)) {
+        if (err instanceof Error && (!retryOpts.isRetryable || retryOpts.isRetryable(err))) {
```

Comment:
i would cast error or expand the type on line 30 rather than adding this to the check. As written in your update this will ignore the retryOpts if the function throws a non-error which i don't think is desirable